packages/core/src/index.ts

```diff
@@ -91,6 +91,18 @@ export function useRef<Value>(val: Value): { current: Value } {
   return useState({ current: val })[0];
 }
 
+export function useReadline(
+  userHandler: (rl: InquirerReadline) => void
+) {
+  const rl = sessionRl;
+
+  if (!rl) {
+    throw new Error('useReadline must be used within a prompt');
+  }
+
+  userHandler(rl);
```

Comment:
Let's just have this hook return the readline; then in the prompt it'll just be:

```js
const rl = useReadline();

useEffect(() => {
  if (!waitForUseInput) {
    startExternalEditor();
  }
}, []);
```